src/components/MultiHopTrade/components/TradeInput/components/ConfirmSummary.tsx

```diff
@@ -101,49 +92,32 @@ export const ConfirmSummary = ({
   const { priceImpactPercentage } = usePriceImpact(activeQuote)
   const walletSupportsBuyAssetChain = useWalletSupportsChain(buyAsset.chainId, wallet)
 
-  const userAddress = useMemo(() => {
-    if (!initialSellAssetAccountId) return ''
-
-    return fromAccountId(initialSellAssetAccountId).account
-  }, [initialSellAssetAccountId])
-
-  const { data: _isSmartContractSellAddress, isLoading: isSellAddressByteCodeLoading } =
-    useIsSmartContractAddress(userAddress, sellAsset.chainId)
-
   const { data: _isSmartContractReceiveAddress, isLoading: isReceiveAddressByteCodeLoading } =
     useIsSmartContractAddress(receiveAddress ?? '', buyAsset.chainId)
 
-  const disableSmartContractSwap = useMemo(() => {
-    // Swappers other than THORChain shouldn't be affected by this limitation
-    if (activeSwapperName !== SwapperName.Thorchain) return false
+  const disableThorNativeSmartContractReceive = useMemo(() => {
```

Comment:
blocking: we should only set this to true if the active swapper is thorchain

suggestion: current logic feels a bit messy to me, just check explicit case in question and leverage `loading` state for remaining disable state.
```ts
  const disableThorNativeSmartContractReceive = useMemo(() => {
    // THORChain is only affected by the sc limitation for native EVM receives
    // https://dev.thorchain.org/protocol-development/chain-clients/evm-chains.html#admonition-warning
    if (
      activeSwapperName === SwapperName.Thorchain &&
      _isSmartContractReceiveAddress &&
      isEvmChainId(buyAsset.chainId) &&
      buyAsset.assetId === buyAssetFeeAsset?.assetId
    )
      return true

    return false
  }, [
    activeSwapperName,
    buyAssetFeeAsset,
    _isSmartContractReceiveAddress,
    buyAsset.chainId,
    buyAsset.assetId,
  ])
```

Reply:
Nice catch! Looks like this has gone one too deep in the cleanup indeed, removing THOR swapper checks.

Also agree assuming disabled on loading state was probably too much and realistically at runtime, we shouldn't need this (i.e a quote will take much longer to get, then it takes for sc checks to finish loading).

d48852bea4